exporters/prometheus/exporter.go

```diff
@@ -122,16 +123,18 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 		}
 	}
 
-	c.createTargetInfoOnce.Do(func() {
-		// Resource should be immutable, we don't need to compute again
-		targetInfo, err := c.createInfoMetric(targetInfoMetricName, targetInfoDescription, metrics.Resource)
+	c.mu.Lock()
+	if c.targetInfo == nil && !c.disableTargetInfo {
+		targetInfo, err := createInfoMetric(targetInfoMetricName, targetInfoDescription, metrics.Resource)
 		if err != nil {
 			// If the target info metric is invalid, disable sending it.
 			otel.Handle(err)
 			c.disableTargetInfo = true
 		}
 		c.targetInfo = targetInfo
-	})
+	}
+	c.mu.Unlock()
```

Comment:
This would not be called if `otel.Handle(err)` panics. I think we should do something like:

```go
func () {
  c.mu.Lock()
  defer c.mu.Unlock()
  // critical section
}()
```

Reply:
I guess if otel.Handle(err) panics that we don't really care if it would not be called?

Reply:
NVM updated.